pkg/wwan/usr/bin/wwan-loc.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/sh
+# shellcheck disable=SC2039
+# shellcheck disable=SC2155
+
+kill_process_tree() {
+  local parent=$1 child
+  for child in $(ps -o ppid= -o pid= | awk "\$1==$parent {print \$2}"); do
+    kill_process_tree $child
+  done
+  kill $parent
+}
+
+# Function keeps publishing location updates to /run/wwan/location.json
+location_tracking() {
+  local CDC_DEV="$1"
+  local PROTOCOL="$2"
+  local OUTPUT_FILE="$3"
+  local MAX_RETRIES=20
```

Comment:
Under what condition(s) will we hit MAX_RETRIES and have the script exit? Do we need to log something in such a case?

Reply:
Actually I wasn't yet able to get the inside of the loop to fail and repeat. Even if the modem does not have GNSS receiver, it will still behave just like a device without GNSS signal - the qmicli command does not return any errors, just hangs without streaming any output. Still I would keep it inside of the loop so that we restart location tracking if something bad happens that I haven't encountered yet. On a second though I think it can be an infinite loop (which is killed when location tracking is disabled), no need to count and limit retries (PR updated).

Reply:
OK